guacamole-common/src/main/java/org/apache/guacamole/protocol/ConfiguredGuacamoleSocket.java

```diff
@@ -184,6 +199,20 @@ public ConfiguredGuacamoleSocket(GuacamoleSocket socket,
                     "image",
                     info.getImageMimetypes().toArray(new String[0])
                 ));
+        
+        // Protocol version 1.1.0 and higher options
+        
+        if (protocol.atLeast(GuacamoleProtocolVersion.VERSION_1_1_0)) {
```

Comment:
Thoughts on providing some sort of capability-style query rather than requiring comparisons on version number? What you've got here will clearly work fine for now, but will get cumbersome as new features are introduced.

Something like `protocol.isSupported(GuacamoleProtocolCapability.HANDSHAKE_TIMEZONE)` would remove the need for repeated version comparison logic here or elsewhere, and would allow us to define what features are supported by what versions in a more readable way.

Reply:
Yeah, I like this idea.  I'm guessing another enum (or class?) would be required to implement this?  Or is there a way that I should try to do it within the current enum?

Reply:
Took a stab at an implementation of this with a new `GuacamoleProtocolCapability` enum.  Let me know if it looks sane or if I should go another direction.